code/SiteConfig.php

```diff
@@ -317,18 +288,13 @@ public function CMSEditLink(): ?string
     /**
      * Get the current sites SiteConfig, and creates a new one through
      * {@link make_site_config()} if none is found.
-     *
-     * @return SiteConfig
      */
-    public static function current_site_config()
+    public static function current_site_config(): SiteConfig
     {
         $siteConfig = DataObject::get_one(SiteConfig::class);
         if (!$siteConfig) {
             $siteConfig = SiteConfig::make_site_config();
         }
-
-        static::singleton()->extend('updateCurrentSiteConfig', $siteConfig);
-
```

Comment:
Removing this because we no longer call `SiteConfig::current_site_config()` in the admin to get the config we want to edit. Instead, this same logic is duplicated there (wtihout the extension hook).

I've checked and nothing in core or supported modules (including fluent and subsites) uses this hook (and [according to GitHub](https://github.com/search?q=updateCurrentSiteConfig&type=code) nor does anything else in the universe) and I also checked with our in-house development teams who weren't able to find any usage of it in their private repositories either. Seems perfectly safe to remove - but I'll mention it in the changelog to be cover our bases.